subgraphs/rocket-pool/src/entityUpdates/financialMetrics.ts

```diff
@@ -0,0 +1,422 @@
+import { Address, BigInt, ethereum, log } from "@graphprotocol/graph-ts";
+import { bigIntToBigDecimal } from "../utils/numbers";
+import { getOrCreateProtocol } from "../entities/protocol";
+import { getOrCreatePool } from "../entities/pool";
+import {
+  FinancialsDailySnapshot,
+  PoolDailySnapshot,
+  PoolHourlySnapshot,
+} from "../../generated/schema";
+import {
+  BIGDECIMAL_ZERO,
+  BIGINT_ZERO,
+  ETH_ADDRESS,
+  RETH_ADDRESS,
+  SECONDS_PER_DAY,
+  SECONDS_PER_HOUR,
+  RPL_ADDRESS,
+  BIGDECIMAL_HALF,
+} from "../utils/constants";
+import { getOrCreateToken } from "../entities/token";
+import { getOrCreateMinipool } from "../entities/Minipool";
+
+const PROTOCOL_ID = RETH_ADDRESS;
+
+export function updateMinipoolTvlandRevenue(
+  block: ethereum.Block,
+  stakingAmount: BigInt,
+  slashingAmount: BigInt,
+  totalRewardEth: BigInt,
+  miniPoolAddress: string
+): void {
+  const Minipool = getOrCreateMinipool(
+    block.number,
+    block.timestamp,
+    miniPoolAddress
+  );
+  const pool = getOrCreatePool(block.number, block.timestamp);
+  Minipool.inputTokenBalances = [
+    Minipool.inputTokenBalances[0].plus(stakingAmount),
+  ];
+  Minipool.totalValueLockedUSD = bigIntToBigDecimal(
+    Minipool.inputTokenBalances[0]
+  ).times(
+    getOrCreateToken(Address.fromString(ETH_ADDRESS), block.number)
+      .lastPriceUSD!
+  );
+
+  Minipool.slashAmount = Minipool.slashAmount.plus(slashingAmount);
+
+  if (totalRewardEth > BIGINT_ZERO) {
+    log.warning("[getOrCreatePool] TotalRewardEth nonzero, updating revenue", [
+      totalRewardEth.toString(),
+    ]);
+    let ratio = Minipool.inputTokenBalances[0].div(pool.inputTokenBalances[0]);
+    let miniPoolRewardEth = ratio.times(totalRewardEth);
+    let miniPoolRewardUSD = bigIntToBigDecimal(miniPoolRewardEth).times(
+      getOrCreateToken(Address.fromString(ETH_ADDRESS), block.number)
+        .lastPriceUSD!
+    );
+    Minipool.cumulativeTotalRevenueUSD = miniPoolRewardUSD;
+    Minipool.cumulativeProtocolSideRevenueUSD = miniPoolRewardUSD.times(
+      BIGDECIMAL_HALF.minus(BIGDECIMAL_HALF.times(Minipool.comissionRate))
+    );
+  }
+  Minipool.save();
+}
+
+export function updateProtocolAndPoolTvl(
+  block: ethereum.Block,
+  amount: BigInt,
+  rewardAmount: BigInt
+): void {
+  const pool = getOrCreatePool(block.number, block.timestamp);
+  const protocol = getOrCreateProtocol();
+
+  let rewardTokens: string[] = [
+    getOrCreateToken(Address.fromString(RPL_ADDRESS), block.number).id,
+  ];
+
+  log.error("[updateProtocolAndPoolTvl] reward tokens check: {}", [
+    rewardTokens[0],
+  ]);
+  pool.rewardTokens = rewardTokens;
+  pool.save();
+  // Pool
+  let inputTokenBalances: BigInt[] = [];
+  inputTokenBalances.push(pool.inputTokenBalances[0].plus(amount));
+  inputTokenBalances.push(pool.inputTokenBalances[1].plus(rewardAmount));
```

Comment:
These amounts will be referring to the wrong token, as per what was said on one of the previous commments

Reply:
swapped